lib/log.bash

```diff
@@ -4,45 +4,56 @@ export BASH_IT_LOG_LEVEL_ERROR=1
 export BASH_IT_LOG_LEVEL_WARNING=2
 export BASH_IT_LOG_LEVEL_ALL=3
 
+function _has_colors()
+{
+    # Check that stdout is a terminal
+    test -t 1 || return 1
+
+    ncolors=$(tput colors)
+    test -n "$ncolors" && test "$ncolors" -ge 8 || return 1
+    return 0
+}
+
 function _log_general()
 {
-  _about 'Internal function used for logging'
-  _param '1: color of the message'
-  _param '2: message to log'
-  _group 'log'
+  about 'Internal function used for logging, uses BASH_IT_LOG_PREFIX as a prefix'
+  param '1: color of the message'
+  param '2: message to log'
+  group 'log'
 
-  _has_colors && echo -e "$1$2${echo_normal}" || echo -e "$2"
+  message=${BASH_IT_LOG_PREFIX}$2
```

Comment:
I just noticed this when running another test... With this order, the messages have the component first, then the log level (DEBUG, WARN, ...), then the message:

```bash
core: main: DEBUG: Loading libraries(except appearance)...
lib: composure: DEBUG: Loading library file...
lib: helpers: DEBUG: Loading library file...
lib: history: DEBUG: Loading library file...
lib: log: DEBUG: Loading library file...
lib: preexec: DEBUG: Loading library file...
lib: preview: DEBUG: Loading library file...
lib: search: DEBUG: Loading library file...
lib: utilities: DEBUG: Loading library file...
```

I think it would be better to have the log level first, then the component, then the message...

```bash
DEBUG: core: main: Loading libraries(except appearance)...
DEBUG: lib: composure: Loading library file...
DEBUG: lib: helpers: Loading library file...
DEBUG: lib: history: Loading library file...
DEBUG: lib: log: Loading library file...
DEBUG: lib: preexec: Loading library file...
DEBUG: lib: preview: Loading library file...
DEBUG: lib: search: Loading library file...
DEBUG: lib: utilities: Loading library file...
```

Thoughts?

Reply:
It does look better, and I changed it!